src/components/vm/consoles/vnc.tsx

```diff
@@ -639,15 +642,18 @@ export const VncMissing = ({
 }) => {
     const [inProgress, setInProgress] = useState(false);
 
-    function add_vnc() {
+    async function add_vnc() {
         setInProgress(true);
-        domainAttachVnc(vm, { })
-                .catch(ex => onAddErrorNotification({
-                    text: cockpit.format(_("Failed to add VNC to VM $0"), vm.name),
-                    detail: ex.message,
-                    resourceId: vm.id,
-                }))
-                .finally(() => setInProgress(false));
+        try {
+            await virtXmlAdd(vm, "graphics", { type: "vnc" });
+        } catch (ex) {
+            onAddErrorNotification({
+                text: cockpit.format(_("Failed to add VNC to VM $0"), vm.name),
+                detail: String(ex),
+                resourceId: vm.id,
```

Comment:
These 5 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2334-da07d010-20250929-141718-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)